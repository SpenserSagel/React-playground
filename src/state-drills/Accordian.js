import React from 'react';

class Accordian extends React.Component{

    static defaultProps = {
        sections : []
    };
    state = {
        currentIndex : -1
    };

    handleClick(index){
        if(this.state.currentIndex===index){
            this.setState({currentIndex:-1});
        }
        else this.setState({currentIndex:index});
    }

    renderButtons(){
        let buttons = [];
        if(!!this.props.sections.length){
            buttons = this.props.sections.map((section,index) =>{
                if(this.state.currentIndex===index){return(
                    <li key={index}>
                        <button key={index} onClick={()=>this.handleClick(index)}>{section.title}</button>
                        <p>{section.content}</p>
                    </li>
                )}
                return(
                    <li key={index}>
                        <button key={index} onClick={()=>this.handleClick(index)}>{section.title}</button>
                    </li>
                )
            });
        }
        else buttons = <li></li>;
        return buttons;
    }

    render(){
        return(
            <ul>
                {this.renderButtons()}
            </ul>
        );
    }
}

export default Accordian;