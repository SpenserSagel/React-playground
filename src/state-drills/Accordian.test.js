import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian.js';

describe('accordian test', () =>{
    const sectionsProp = [
        {
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ];

    it('renders empty by default', () => {
        const wrapper = shallow(<Accordian/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders closed sections by default', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('opens a section on button click', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp}/>);
        wrapper.find('button').at(0).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('closes a section when another is opened', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
        
    });
})