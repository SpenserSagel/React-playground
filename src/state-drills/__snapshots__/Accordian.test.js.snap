// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordian test closes a section when another is opened 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`accordian test opens a section on button click 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`accordian test renders closed sections by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`accordian test renders empty by default 1`] = `ShallowWrapper {}`;
